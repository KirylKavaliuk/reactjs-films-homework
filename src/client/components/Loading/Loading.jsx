import React from 'react';

import classNames from 'classnames';

import styles from './Loading.scss';

const Loading = ({ className }) => (
  <div className={
    classNames(
      styles.loading,
      className,
    )
    }>
    <div className={ styles.icons }>
      <div className={ styles.icon }>
        <div className={ styles.circle }/>
      </div>
      <div className={ styles.icon }>
        <div className={ styles.circle }/>
      </div>
      <div className={ styles.icon }>
        <div className={ styles.circle }/>
      </div>
    </div>
    <span className={ styles.text }>Loading</span>
  </div>
);

export default Loading;
