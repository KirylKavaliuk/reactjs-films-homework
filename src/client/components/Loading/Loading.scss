@import 'styles/palette';
@import 'styles/animations';

.loading {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 30px 0 0;
}

.icons {
  display: flex;
  height: 13px;
  justify-content: center;
  align-items: center;
}

.icon {
  display: flex;
  width: 25px;
  height: 25px;
  align-items: center;
  justify-content: center;
}

.circle {
  width: 13px;
  height: 13px;
  margin: 5px;
  animation: changeSizes 0.6s infinite ease alternate;
  border-radius: 30px;
  background: $light_blue;
}

.icon:nth-child(1) .circle {
  animation-delay: 0.2s;
}

.icon:nth-child(2) .circle {
  animation-delay: 0.4s;
}

.icon:nth-child(3) .circle {
  animation-delay: 0.6s;
}

.text {
  box-sizing: border-box;
  padding: 10px 0 0;
  letter-spacing: 1px;
  text-transform: uppercase;
  color: $grey;
  font-size: 18px;
}
