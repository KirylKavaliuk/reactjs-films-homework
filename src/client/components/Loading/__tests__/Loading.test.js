import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Loading from '../Loading';

it('<Loading/>', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Loading/>);

  const tree = renderer.getRenderOutput();

  expect(tree).toMatchSnapshot();
});

it('<Loading/>', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Loading className='Component__class-name__hash'/>);

  const tree = renderer.getRenderOutput();

  expect(tree).toMatchSnapshot();
});
