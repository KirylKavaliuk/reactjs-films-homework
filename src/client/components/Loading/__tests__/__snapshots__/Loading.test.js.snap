// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders loading correctly 1`] = `
<div
  className="Loading__loading___2_FrG"
>
  <div
    className="Loading__icons___1uA0Q"
  >
    <div
      className="Loading__icon___2WVPJ"
    >
      <div
        className="Loading__circle___14YyA"
      />
    </div>
    <div
      className="Loading__icon___2WVPJ"
    >
      <div
        className="Loading__circle___14YyA"
      />
    </div>
    <div
      className="Loading__icon___2WVPJ"
    >
      <div
        className="Loading__circle___14YyA"
      />
    </div>
  </div>
  <span
    className="Loading__text___13r1Y"
  >
    Loading
  </span>
</div>
`;

exports[`renders loading correctly 2`] = `
<div
  className="Loading__loading___2_FrG Component__class-name__hash"
>
  <div
    className="Loading__icons___1uA0Q"
  >
    <div
      className="Loading__icon___2WVPJ"
    >
      <div
        className="Loading__circle___14YyA"
      />
    </div>
    <div
      className="Loading__icon___2WVPJ"
    >
      <div
        className="Loading__circle___14YyA"
      />
    </div>
    <div
      className="Loading__icon___2WVPJ"
    >
      <div
        className="Loading__circle___14YyA"
      />
    </div>
  </div>
  <span
    className="Loading__text___13r1Y"
  >
    Loading
  </span>
</div>
`;
